### ISSUE TITLE
False-Positive involving overridden and overlapping interfaces (and specific interface names)

### ISSUE THREAD

**Issue by blablubbabc** [typetools/checker-framework#5069]
I am using the Gradle plugin (v 0.6.8) to invoke the CheckerFramework (v 3.21.2).

I briefly tried to come up with a more minimal example, but it appears that the condition under which this issue can be reproduced is quite fragile. Even changes to the interface names (maybe affecting their alphabetical ordering) have an effect on whether the issue can be reproduced.  
Reproduction example:
```
public interface HasMaybeName {
	public @Nullable String getName();
}

public interface HasName extends HasMaybeName {
	@Override
	public @NonNull String getName();
}

// If I rename this to 'ExtendedHasMaybeName' instead of 'OExtendedHasMaybeName'
// I cannot not reproduce the issue anymore.
public interface OExtendedHasMaybeName extends HasMaybeName {
	@Override
	public @Nullable String getName();
}

public interface Thing extends HasName, OExtendedHasMaybeName {
}

public class ActualThing implements Thing {
	@Override
	public @NonNull String getName() {
		return "";
	}
}

public class Test {

	public static Thing getThing() {
		return new ActualThing();
	}

	public static void foo() {
		Thing thing = getThing();
		@NonNull String name = thing.getName(); // Error
		doSth(name);
		doSth(thing.getName()); // Error
	}

	public static void doSth(String name) {
		assert System.out != null;
		System.out.println(name);
	}
}
```

CF reports the following errors:
```
[assignment] incompatible types in assignment.

		@NonNull String name = thing.getName(); // Error
		                                    ^
  found   : @Initialized @Nullable String

  required: @UnknownInitialization @NonNull String
```
```
[argument] incompatible argument for parameter name of doSth.

		doSth(thing.getName()); // Error
		                   ^
  found   : @Initialized @Nullable String

  required: @Initialized @NonNull String
```

**Comment by smillst** [comment]
Below is a repro of this bug.  In this example, changing the order of the interfaces cause the false positive to disappear.   The problem is that the Checker Framework uses the signature of the method it finds first to type check even if there may be more than one method.  Instead the Checker Framework should look at the methods in all interfaces.

```
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullable;

interface NullableInterface {
  @Nullable String method();
}

// If NullableSubInterface appears in the source before NonNullSubInterface, then the false positive
// below will not occur.
interface NonNullSubInterface extends NullableInterface {
  @Override
  @NonNull String method();
}

interface NullableSubInterface extends NullableInterface {
  @Override
  @Nullable String method();
}

interface BothInterface extends NonNullSubInterface, NullableSubInterface {}

public class Test {
  public static void foo(BothInterface thing) {
    // This should be legal because any class that implements BothInterface must implement method
    // with a signature that is a subsignature of both NonNullSubInterface and NullableSubInterface.
    @NonNull String name = thing.method(); // Error
  }
}

class BothClass1 implements BothInterface {
  @Override
  // :: error: (override.return)
  public @Nullable String method() {
    return null;
  }
}

class BothClass2 implements BothInterface {
  @Override
  public @NonNull String method() {
    return "";
  }
}
```
